std/datetime/date.d

```diff
@@ -6594,12 +6594,12 @@ public:
             $(REF DateTimeException,std,datetime,date) if the given day is an
             invalid day of the year.
       +/
-    @property void dayOfYear(int day) @safe pure
+    @property void dayOfYear(int day) @safe pure nothrow @nogc
     {
         immutable int[] lastDay = isLeapYear ? lastDayLeap : lastDayNonLeap;
 
         if (day <= 0 || day > (isLeapYear ? daysInLeapYear : daysInYear))
-            throw new DateTimeException("Invalid day of the year.");
+            assert(0, "Invalid day of the year.");
```

Comment:
1) There's also the trick/proposal to use Singleton's for exceptions (e.g. https://github.com/dlang/druntime/pull/1710/files).

2) Does anyone know the exact status of [DIP1008](https://github.com/dlang/DIPs/blob/master/DIPs/DIP1008.md)? The DIP status was set to " Post-Preliminary Review Round 1" [26 days ago](https://github.com/dlang/DIPs/commit/8051eec15b26fb52c759cdf34bda2cd69b825800).